jetty-setuid/jetty-setuid-native/src/main/resources/org_eclipse_jetty_setuid_SetUID.c

```diff
@@ -58,7 +58,12 @@ Java_org_eclipse_jetty_setuid_SetUID_setumask (JNIEnv * env, jclass j, jint mask
 JNIEXPORT jint JNICALL 
 Java_org_eclipse_jetty_setuid_SetUID_setgid (JNIEnv * env, jclass j, jint gid)
 {
-    return((jint)setgid((gid_t)gid));
+    int result = setgroups(0, NULL);
```

Comment:
I can see this fixes the problem,  but all the other methods here are pretty much 1:1 with the java:C, so it is a bit strange that `setgid` does a `setgroups` as well.   Would it not be better to add API for setGroups and then we can call that from the java?

Reply:
Yeah, I was guessing someone might say that, and I'm happy to do it this way if you prefer.

Do you need to be able to call `setgroups` with actual arguments do you think, or will only supporting clearing it as in this example be enough?

In either case, I'll aim to get back to it soon.